token/cli/src/main.rs

```diff
@@ -1703,22 +1713,32 @@ fn main() {
             SubCommand::with_name("close")
                 .about("Close a token account")
                 .arg(
-                    Arg::with_name("account")
+                    Arg::with_name("token")
                         .validator(is_valid_pubkey)
-                        .value_name("TOKEN_ACCOUNT_ADDRESS")
+                        .value_name("TOKEN_ADDRESS")
                         .takes_value(true)
                         .index(1)
-                        .required(true)
-                        .help("The address of the token account to close"),
+                        .required_unless("address")
+                        .help("Token to close. To close a specific account, use the `--address` parameter instead"),
                 )
                 .arg(
-                    Arg::with_name("destination")
+                    Arg::with_name("recipient")
+                        .long("recipient")
                         .validator(is_valid_pubkey)
                         .value_name("REFUND_ACCOUNT_ADDRESS")
                         .takes_value(true)
-                        .index(2)
                         .help("The address of the account to receive remaining SOL [default: --owner]"),
                 )
+                .arg(
+                    Arg::with_name("address")
+                        .long("address")
+                        .validator(is_valid_pubkey)
+                        .value_name("TOKEN_ACCOUNT_ADDRESS")
+                        .takes_value(true)
+                        .required_unless("token")
```

Comment:
Does this include `--address` in the error message if the `token` positional is omitted?  I think ideally we only suggest the ATA usage and they can `--help` to find `--address`.  I'm not sure if we can trick `clap` into that though?

Reply:
Currently, yes, and I agree it's not ideal. Let me try some asymmetric options to see if I can hide `--address` from that error msg...

Reply:
K, I think this is nicer:

```
$ spl-token balance
error: The following required arguments were not provided:
    <TOKEN_ADDRESS>

USAGE:
    spl-token balance <TOKEN_ADDRESS> --config <PATH>
```

```
$ spl-token balance 2ywp5tbpeeEUAM2PFP6NPtvovyFVnjX3qNfrd13S3HmG --address CYRJWqiSjLitBAcRxPvWpgX3s5TvmN2SuRY3eEYypFvT
error: The argument '<TOKEN_ADDRESS>' cannot be used with '--address <TOKEN_ACCOUNT_ADDRESS>'

USAGE:
    spl-token balance <TOKEN_ADDRESS> --address <TOKEN_ACCOUNT_ADDRESS> --config <PATH>
```

and `spl-token balance <TOKEN_ADDRESS>` and `spl-token balance --address <TOKEN_ACCOUNT_ADDRESS>` just work

Reply:
Lovely!  Thanks for digging in on that!